CRM/Case/Form/Activity.php

```diff
@@ -272,12 +272,13 @@ public function buildQuickForm() {
     $this->assign('urlPath', 'civicrm/case/activity');
 
     $encounterMediums = CRM_Case_PseudoConstant::encounterMedium();
-    // Fixme: what's the justification for this? It seems like it is just re-adding an option in case it is the default and disabled.
-    // Is that really a big problem?
     if ($this->_activityTypeFile == 'OpenCase') {
-      $this->_encounterMedium = CRM_Core_DAO::getFieldValue('CRM_Activity_DAO_Activity', $this->_activityId,
-        'medium_id'
-      );
+      if ($this->_action == CRM_Core_Action::UPDATE) {
+        $this->getElement('activity_date_time')->freeze();
+      }
+      // Fixme: what's the justification for this? It seems like it is just re-adding an option in case it is the default and disabled.
+      // Is that really a big problem?
+      $this->_encounterMedium = CRM_Core_DAO::getFieldValue('CRM_Activity_DAO_Activity', $this->_activityId, 'medium_id');
       if (!array_key_exists($this->_encounterMedium, $encounterMediums)) {
         $encounterMediums[$this->_encounterMedium] = CRM_Core_OptionGroup::getLabel('encounter_medium',
           $this->_encounterMedium,
```

Comment:
Should we also remove the param `activity_date_time_time` from https://github.com/civicrm/civicrm-core/blob/master/CRM/Case/Form/Activity.php#L408? Currently, on saving a case activity, I get a notice error as follow -

![image](https://user-images.githubusercontent.com/5929648/50476936-2b041680-09f0-11e9-97c9-23ded9df2800.png)

Other changes seems to work fine on my testing 👍

Reply:
The whole line 408 and the comment above can be removed. They are not required anymore.